tensorflow_addons/seq2seq/attention_wrapper_test.py

```diff
@@ -27,9 +27,9 @@
 from tensorflow_addons.seq2seq import attention_wrapper as wrapper
 from tensorflow_addons.seq2seq import basic_decoder
 from tensorflow_addons.seq2seq import sampler as sampler_py
-from tensorflow.python import keras
-from tensorflow.python.eager import context
 
+# TODO: Find public API alternatives to these
+from tensorflow.python import keras
 from tensorflow.python.keras import initializers
```

Comment:
These look like there should be an easy replacement, but I was getting test failures when replacing them. Not sure but maybe related to how they're being used. For example:

```
if cell._attention_layers is not None:
                for layer in cell._attention_layers:
                    layer.kernel_initializer = initializers.glorot_uniform(
                        seed=1337)
```
Replacing that with tf.keras.intializers.glort_uniform did fails.

Reply:
hum, "tf.keras.intializers.glorot_uniform" has a typo in it, should be "tf.keras.initializers.glorot_normal". Can you attach the test failure log?

Reply:
Runnning` tf.keras.initializers.glorot_uniform` or `tf.keras.initializers.glorot_normal`:

Multiple failures for array comparisons.. example:
```
======================================================================
FAIL: testLuongScaled (__main__.AttentionWrapperTest)
testLuongScaled (__main__.AttentionWrapperTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1031, in decorated
    f(self, *args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/seq2seq/attention_wrapper_test.runfiles/__main__/tensorflow_addons/seq2seq/attention_wrapper_test.py", line 611, in testLuongScaled
    create_attention_kwargs=create_attention_kwargs)
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/seq2seq/attention_wrapper_test.runfiles/__main__/tensorflow_addons/seq2seq/attention_wrapper_test.py", line 240, in _testWithAttention
    create_attention_kwargs=create_attention_kwargs)
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/seq2seq/attention_wrapper_test.runfiles/__main__/tensorflow_addons/seq2seq/attention_wrapper_test.py", line 410, in _testWithMaybeMultiAttention
    expected_final_output, final_output_info)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/util/nest.py", line 509, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/seq2seq/attention_wrapper_test.runfiles/__main__/tensorflow_addons/seq2seq/attention_wrapper_test.py", line 192, in assertAllCloseOrEqual
    x, y, atol=1e-3, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1073, in decorated
    return f(*args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 2303, in assertAllClose
    self._assertAllCloseRecursive(a, b, rtol=rtol, atol=atol, msg=msg)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 2272, in _assertAllCloseRecursive
    (path_str, path_str, msg)))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 2207, in _assertArrayLikeAllClose
    a, b, rtol=rtol, atol=atol, err_msg="\n".join(msgs), equal_nan=True)
  File "/usr/local/lib/python2.7/dist-packages/numpy/testing/nose_tools/utils.py", line 1396, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/local/lib/python2.7/dist-packages/numpy/testing/nose_tools/utils.py", line 779, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-06, atol=0.001
Mismatched value: a is different from b. 
not close lhs = -0.06124732
not close rhs = 0.247038751841
not close dif = 0.308286071841
not close tol = 0.00100024703875
dtype = float64, shape = ()
(mismatch 100.0%)
 x: array(-0.061247)
 y: array(0.247039, dtype=float32)

----------------------------------------------------------------------
Ran 36 tests in 38.762s
FAILED (failures=8, skipped=2)
```

Reply:
Humm, that's weird, unless it changes the RNG for weight initialize, otherwise I don't see why it could cause any numerical difference. I will do some debug for this.